lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,285 @@
+#
+# NXAPI implementation of Router OSPF Area class
+#
+# June 2016, Sai Chintalapudi
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+require_relative 'node_util'
+require_relative 'router_ospf'
+require_relative 'router_ospf_vrf'
+
+module Cisco
+  # node_utils class for ospf_area
+  class RouterOspfArea < NodeUtil
+    attr_reader :router, :vrf, :area_id
+
+    def initialize(ospf_router, vrf_name, area_id)
+      fail TypeError unless ospf_router.is_a?(String)
+      fail TypeError unless vrf_name.is_a?(String)
+      fail TypeError unless area_id.is_a?(String)
+      fail ArgumentError unless ospf_router.length > 0
+      fail ArgumentError unless vrf_name.length > 0
+      fail ArgumentError unless area_id.length > 0
+
+      # check the area_id is a proper ipv4 address
+      fail ArgumentError if (begin
+                               IPAddr.new(area_id)
+                             rescue
+                               nil
+                             end).nil?
+      fail ArgumentError unless IPAddr.new(area_id).ipv4?
```

Comment:
Area IDs can be simple integers, however, the backend platform code converts the intgeter to ipaddr and the show command only shows them as ipv4 address. 
for ex:
agent-lab13-nx(config)# router ospf abcd
agent-lab13-nx(config-router)# area 123 stub
agent-lab13-nx(config-router)# sh running-config ospf

!Command: show running-config ospf
!Time: Wed Jun  8 10:50:52 2016

version 7.0(3)I3(1)
feature ospf

router ospf abcd
  area 0.0.0.123 stub

If we do allow, we need to keep converting the integers to ipv4 addresses and use them internally. I thought it is unnecessary to allow 2 types of inputs when the output is always ipv4. So basically forcing the user to give us the actual ipv4 address. Let me know if you disagree.